packages/backend-common/src/database/connectors/mysql.ts

```diff
@@ -229,3 +238,243 @@ export const mysqlConnector: DatabaseConnector = Object.freeze({
+    private readonly prefix: string,
+  ) {}
+
+  async getClient(
+    pluginId: string,
+    deps?: {
+      lifecycle: LifecycleService;
+      pluginMetadata: PluginMetadataService;
+    },
+  ): Promise<Knex> {
+    const pluginConfig = new ConfigReader(
+      this.getConfigForPlugin(pluginId) as JsonObject,
+    );
+
+    const databaseName = this.getDatabaseName(pluginId);
+    if (databaseName && this.getEnsureExistsConfig(pluginId)) {
+      try {
+        await mysqlConnector.ensureDatabaseExists!(pluginConfig, databaseName);
+      } catch (error) {
+        throw new Error(
+          `Failed to connect to the database to make sure that '${databaseName}' exists, ${error}`,
+        );
+      }
+    }
+
+    const pluginDivisionMode = this.getPluginDivisionModeConfig();
+    if (pluginDivisionMode !== 'database') {
+      throw new Error(
+        `The MySQL driver does not suppoert plugin division mode '${pluginDivisionMode}'`,
+      );
+    }
+
+    const databaseClientOverrides = mergeDatabaseConfig(
+      {},
+      this.getDatabaseOverrides(pluginId),
+    );
+
+    const client = mysqlConnector.createClient(
+      pluginConfig,
+      databaseClientOverrides,
+      deps,
+    );
+
+    return client;
+  }
+
+  async dropDatabase(...databaseNames: string[]): Promise<void> {
+    return await dropMysqlDatabase(this.config, ...databaseNames);
+  }
+
+  /**
+   * Provides the canonical database name for a given plugin.
+   *
+   * This method provides the effective database name which is determined using
+   * global and plugin specific database config. If no explicit database name,
+   * this method will provide a generated name which is the pluginId prefixed
+   * with 'backstage_plugin_'.
+   *
+   * @param pluginId - Lookup the database name for given plugin
+   * @returns String representing the plugin's database name
+   */
+  private getDatabaseName(pluginId: string): string | undefined {
+    const connection = this.getConnectionConfig(pluginId);
+    const databaseName = (connection as Knex.ConnectionConfig)?.database;
+    return databaseName ?? `${this.prefix}${pluginId}`;
+  }
+
+  /**
+   * Provides the client type which should be used for a given plugin.
+   *
+   * The client type is determined by plugin specific config if present.
+   * Otherwise the base client is used as the fallback.
+   *
+   * @param pluginId - Plugin to get the client type for
+   * @returns Object with client type returned as `client` and boolean
+   *          representing whether or not the client was overridden as
+   *          `overridden`
+   */
+  private getClientType(pluginId: string): {
+    client: string;
+    overridden: boolean;
+  } {
+    const pluginClient = this.config.getOptionalString(
+      `${pluginPath(pluginId)}.client`,
+    );
+
+    const baseClient = this.config.getString('client');
+    const client = pluginClient ?? baseClient;
+    return {
+      client,
+      overridden: client !== baseClient,
```

Comment:
Feels like we still have some complexity here where we kinda need to go back up? Or is that not the case because reading an override config is much simpler than reading the full config, and doesn't need database specific code?

Reply:
This method is also just unmodified copypasta. In some cases implementations care about this flag, sometimes they don't. Pg does. This needs cleanup later on too